allennlp/data/dataset_readers/dataset_utils/text2sql_utils.py

```diff
@@ -0,0 +1,137 @@
+
+"""
+Utility functions for reading the standardised text2sql datasets presented in
+`"Improving Text to SQL Evaluation Methodology" <https://arxiv.org/abs/1806.09029>`_
+"""
+from typing import List, Dict, Tuple, NamedTuple, Iterable
+import json
+
+from allennlp.common import JsonDict
+
+
+class SqlData(NamedTuple):
+    """
+    A utility class for reading in text2sql data.
+
+    Parameters
+    ----------
+    text : ``List[str]``
+        The tokens in the text of the query.
+    text_with_variables : ``List[str]``
+        The tokens in the text of the query with variables
+        mapped to table names/abstract variables.
+    sql : ``List[str]``
+        The tokens in the SQL query which corresponds to the text.
+    text_variables : ``Dict[str, str]``
+        A dictionary of variables associated with the text, e.g. {"city_name0": "san fransisco"}
+    sql_variables : ``Dict[str, str]``
+        A dictionary of variables associated with the sql query.
+    """
+    text: List[str]
+    text_with_variables: List[str]
+    sql: List[str]
+    text_variables: Dict[str, str]
+    sql_variables: Dict[str, str]
+
+
+def get_tokens(sentence: List[str],
+               sentence_variables: Dict[str, str]) -> List[str]:
+    """
+    Replaces abstract variables in text with their concrete counterparts.
+    """
+    tokens = []
+    for token in sentence:
+        if token not in sentence_variables:
+            tokens.append(token)
+        else:
+            for word in sentence_variables[token].split():
+                tokens.append(word)
+    return tokens
+
+def clean_and_split_sql(sql: str) -> List[str]:
+    """
+    Cleans up and unifies a SQL query. This involves removing uncessary quotes
+    and spliting brackets which aren't formatted consistently in the data.
+    """
+    sql_tokens = []
+    for token in sql.strip().split():
+        token = token.replace('"', "").replace('"', "").replace("%", "")
+        if token.endswith("(") and len(token) > 1:
+            sql_tokens.append(token[:-1])
+            sql_tokens.append(token[-1])
+        else:
+            sql_tokens.append(token)
+    return sql_tokens
+
+
+def process_sql_data_blob(data: JsonDict,
```

Comment:
A docstring explaining the parameters here would be nice.  I think `data` is the whole dataset as formatted by the text-to-sql paper?

Reply:
Oh, no it's not...